### PR TITLE
Migrate beam_io package to Python3.

### PR DESCRIPTION
**Pull request by tneymanov**
I refactored the pylint related changes out into a new commit and added travis changes; however, I forgot to modify the description of the second commit, which now adds only vital manual changes (no pylint). Apologies about that.

When squash merging, it won't matter.

### CODE REVIEW

File: gcp_variant_transforms/beam_io/vcf_header_io_test.py [hunk 1]
```diff
@@ -311,10 +313,10 @@ def test_write_info_number_types(self):
     header = _get_vcf_header_from_lines(self.lines)
     header_fn = WriteVcfHeaderFn('')
     actual = []
-    for info in header.infos.values():
+    for info in list(header.infos.values()):
       actual.append(header_fn._to_vcf_header_line('INFO', info))
     expected = self.lines[:-1]
-    self.assertItemsEqual(actual, expected)
+    self.assertCountEqual(actual, expected)
```
Comment (samanvp) [inline gcp_variant_transforms/beam_io/vcf_header_io_test.py:319]: Why we only are verifying their counts?

File: gcp_variant_transforms/beam_io/vcf_parser.py [hunk 2]
```diff
@@ -152,7 +152,8 @@ def __lt__(self, other):
     other_vars = vars(other)
     for key in sorted(self_vars):
       if self_vars[key] != other_vars[key]:
-        return self_vars[key] < other_vars[key]
+        return (other_vars[key] is not None and
+                (self_vars[key] is None or self_vars[key] < other_vars[key]))
```
Comment (samanvp) [inline gcp_variant_transforms/beam_io/vcf_parser.py:156]: why those None checks are being added here?

File: gcp_variant_transforms/beam_io/vcf_parser_test.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 """Tests for vcf_parser module."""
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/beam_io/vcf_parser_test.py:17]: Remove extra line.

File: gcp_variant_transforms/beam_io/vcfio.py [hunk 4]
```diff
@@ -17,7 +17,7 @@
 The 4.2 spec is available at https://samtools.github.io/hts-specs/VCFv4.2.pdf.
 """
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/beam_io/vcfio.py:20]: Remove extra line.

File: gcp_variant_transforms/beam_io/vcfio.py [hunk 5]
```diff
@@ -87,7 +87,7 @@ def encode(self, variant):
       columns.append(encoded_calls)
     columns = [self._encode_value(c) for c in columns]
 
-    return '\t'.join(columns) + '\n'
+    return ('\t'.join(columns) + '\n').encode('utf-8')
```
Comment (samanvp) [inline gcp_variant_transforms/beam_io/vcfio.py:90]: Please update the type hint of this method.

Also in all the previous files that we replaced str with byte please check the type hints and update them.

File: gcp_variant_transforms/beam_io/vcfio.py [hunk 6]
```diff
@@ -112,7 +112,7 @@ def _encode_variant_info(self, variant):
         and start_0_based + len(variant.reference_bases) != variant.end):
       encoded_infos.append('END=%d' % variant.end)
     # Set all other fields of info.
-    for k, v in variant.info.iteritems():
+    for k, v in list(variant.info.items()):
```
Comment (samanvp) [inline gcp_variant_transforms/beam_io/vcfio.py:115]: Remove `list()`

File: gcp_variant_transforms/beam_io/vcfio.py [hunk 7]
```diff
@@ -280,9 +280,10 @@ def __init__(self,
     self._sample_name_encoding = sample_name_encoding
     self._use_1_based_coordinate = use_1_based_coordinate
 
-  def _read_records(self, (file_path, block)):
+  def _read_records(self, xxx_todo_changeme):
     # type: (Tuple[str, Block]) -> Iterable(Variant)
     """Reads records from `file_path` in `block`."""
+    (file_path, block) = xxx_todo_changeme
```
Comment (samanvp) [inline gcp_variant_transforms/beam_io/vcfio.py:286]: Update the name.

File: gcp_variant_transforms/beam_io/vcfio_test.py [hunk 8]
```diff
@@ -15,7 +15,7 @@
 
 """Tests for vcfio module."""
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/beam_io/vcfio_test.py:18]: Remove extra line.

File: gcp_variant_transforms/beam_io/vcfio_test.py [hunk 9]
```diff
@@ -824,11 +823,11 @@ class VcfSinkTest(unittest.TestCase):
   def setUp(self):
     super(VcfSinkTest, self).setUp()
     self.path = tempfile.NamedTemporaryFile(suffix='.vcf').name
-    self.variants, self.variant_lines = zip(
+    self.variants, self.variant_lines = list(zip(
```
Comment (samanvp) [inline gcp_variant_transforms/beam_io/vcfio_test.py:826]: hmm, does this actually work?

File: gcp_variant_transforms/beam_io/vcfio_test.py [hunk 10]
```diff
@@ -837,17 +836,17 @@ def _assert_variant_lines_equal(self, actual, expected):
     self.assertEqual(len(actual_fields), len(expected_fields))
     self.assertEqual(actual_fields[0], expected_fields[0])
     self.assertEqual(actual_fields[1], expected_fields[1])
-    self.assertItemsEqual(actual_fields[2].split(';'),
+    self.assertCountEqual(actual_fields[2].split(';'),
```
Comment (samanvp) [inline gcp_variant_transforms/beam_io/vcfio_test.py:839]: Again, please confirm these asserts are as strong as the old ones.

File: gcp_variant_transforms/beam_io/vcfio.py (hunk 7 above)
Comment (tneymanov) [reply gcp_variant_transforms/beam_io/vcfio.py]: Oops, Done. now I have to check for this everywhere :'(

File: gcp_variant_transforms/beam_io/vcfio_test.py (hunk 8 above)
Comment (tneymanov) [reply gcp_variant_transforms/beam_io/vcfio_test.py:18]: Done.

File: gcp_variant_transforms/beam_io/vcfio_test.py (hunk 10 above)
Comment (tneymanov) [reply gcp_variant_transforms/beam_io/vcfio_test.py:838]: Yeah, they are identical - https://six.readthedocs.io/#six.assertCountEqual

File: gcp_variant_transforms/beam_io/vcfio.py (hunk 5 above)
Comment (tneymanov) [reply gcp_variant_transforms/beam_io/vcfio.py:89]: SG, but please double check if I miss any.

File: gcp_variant_transforms/beam_io/vcfio.py (hunk 6 above)
Comment (tneymanov) [reply gcp_variant_transforms/beam_io/vcfio.py]: Done.

**Pull request merged**